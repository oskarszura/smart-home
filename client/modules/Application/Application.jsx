import React from 'react';
import PropTypes from 'prop-types';
import AlertPanel from 'modules/AlertPanel';

const Application = props => (
  <div className="application">
    {props.children}
    <AlertPanel />
  </div>
);

Application.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired,
};

export default Application;
