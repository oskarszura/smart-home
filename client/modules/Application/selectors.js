// @flow
export const getIsLoaded = (state: Object): WebSocket => {
  return state.application.isLoaded;
};
