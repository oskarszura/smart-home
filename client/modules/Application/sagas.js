// @flow
import _ from 'lodash';
import { all, put, take, select } from 'redux-saga/effects';
import * as agentSelectors from 'client/models/agents/selectors'
import * as agentActions from 'client/models/agents/actions';
import * as agentConfigActions from 'client/models/agentConfigs/actions';
import * as agentActionTypes from 'client/models/agents/actionTypes';
import * as actions from './actions';

/* eslint-disable import/prefer-default-export */
export function* mount(): Iterable<any> {
  yield all([
    put(agentActions.fetchData()),
  ]);
  yield take([agentActionTypes.DATA_FETCH_SUCCESS]);

  const agents = yield select(agentSelectors.getAgents);
  yield all(
    _.map(agents, agent => put(agentConfigActions.fetchData(agent.id)))
  );

  yield put(actions.loaded());
}
/* eslint-enable import/prefer-default-export */
