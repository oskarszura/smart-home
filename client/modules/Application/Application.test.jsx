import React from 'react';
import { shallow } from 'enzyme';
import Application from './Application';

describe('Application', () => {
  it('should render correctly', () => {
    const component = shallow(
      <Application>
        Content
      </Application>
    );

    expect(component).toMatchSnapshot();
  });
});
