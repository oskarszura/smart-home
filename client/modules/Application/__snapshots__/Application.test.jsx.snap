// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Application should render correctly 1`] = `
<div
  className="application"
>
  Content
  <Connect(AlertPanel) />
</div>
`;
